docker-entrypoint.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+set -e
+
+# Set default registry URI, same as Dockerfile
+REGISTRY_URI="/hyperlane-registry"
+
+# Only update registry if REGISTRY_COMMIT is set
+if [ -n "$REGISTRY_COMMIT" ]; then
+  echo "Updating Hyperlane registry to commit: ${REGISTRY_COMMIT}"
+  OLDPWD=$(pwd)
+  cd "$REGISTRY_URI"
+  git fetch origin "$REGISTRY_COMMIT"
```

Comment:
Hoping we don't end up in a bad rate limiting situation - I've def had my monorepo builds fail in CI before due to rate limiting. But happy to worry about that if/when it happens, and I suppose it'll just restart anyways in k8s bc of the `set -e` if that happens